apps/server/test/integration/auth.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { before, beforeEach } from './helpers';
+
+const context: any = {};
+
+beforeAll(async () => {
+	await before(context);
+	await beforeEach(context);
+});
+
+const getSelfWithHeaders = async (headers = {}) => {
+	return context.http('GET', '/api/v2/whoami', {}, headers);
+};
+
+test('server should associate requests with no credentials with a guest user', async () => {
+	const { response } = await getSelfWithHeaders({});
+	expect(!response.error);
+	expect(response.data.slug).toMatch('user-guest');
+});
+
+test('server should associate requests with an invalid authentication header with a guest user', async () => {
+	const { response } = await getSelfWithHeaders({
+		Authorization: 'Bearer foobar',
+	});
+	expect(response.error);
+	expect(response.data.slug).toMatch('user-guest');
+});
+
+test('server should accept requests containing a valid session ID in the authorization header', async () => {
+	const { response } = await getSelfWithHeaders({
+		Authorization: `Bearer ${context.session.id}`,
+	});
+	expect(!response.error);
+	expect(response.data.slug).not.toMatch('user-guest');
+});
+
+test('server should accept requests containing the session token in the authorization header and the session id in the x-session-id header', async () => {
+	const { response } = await getSelfWithHeaders({
+		Authorization: `Bearer ${context.sessionToken}`,
+		'X-Session-ID': context.session.id,
```

Comment:
Might be overkill, but what happens when only `X-Session-ID` is provided in a request?

Reply:
As of now, when only `X-Session-ID` is provided, the auth middleware will look for the token in the `Authorization` header and find nothing. Therefore, it will not be able to authenticate and should use the guest session. I will add a test case to cover this scenario as well.

Reply:
@joshbwlng, done.